app/domain/authentication/authn_jwt/input_validation/validate_claim_name.rb

```diff
@@ -0,0 +1,55 @@
+module Authentication
+  module AuthnJwt
+    module InputValidation
+      # Validate the claim name value
+      ValidateClaimName ||= CommandClass.new(
+        dependencies: {
+          regexp_class: Regexp,
+          deny_claims_list_value: [],
+          logger: Rails.logger
+        },
+        inputs: %i[claim_name]
+      ) do
+        def call
+          @logger.debug(LogMessages::Authentication::AuthnJwt::ValidatingClaimName.new(@claim_name))
+          validate_claim_name_exists
+          validate_claim_name_value
+          validate_claim_is_allowed
+          @logger.debug(LogMessages::Authentication::AuthnJwt::ValidatedClaimName.new(@claim_name))
+        end
+
+        private
+
+        def validate_claim_name_exists
+          raise Errors::Authentication::AuthnJwt::FailedToValidateClaimMissingClaimName if @claim_name.blank?
+        end
+
+        def validate_claim_name_value
+          return if claim_name_regex.match?(@claim_name)
+
+          raise Errors::Authentication::AuthnJwt::FailedToValidateClaimForbiddenClaimName.new(
+            @claim_name,
+            MANDATORY_CLAIMS_RESOURCE_NAME,
```

Comment:
`MANDATORY_CLAIMS_RESOURCE_NAME` looks unrelated to this great class and in general to the PR.
Why do we couple this pure logic to the resource name.
I think that we should keep it resource independent.
Will Tamir use the same class for mapping validation?

Reply:
Great comment, i will review again this class from the thinking that other component will use it